server/athenian/api/internal/settings.py

```diff
@@ -833,20 +821,19 @@ async def set_release_matches(
             )
         values = []
         for repo in repos:
-            if RepositoryName.from_prefixed(repo).is_logical:
-                if match not in (ReleaseMatch.tag, ReleaseMatch.event):
-                    raise ResponseError(
-                        InvalidRequestError(
-                            f"{pointer_root}.{match}",
-                            detail=(
-                                f"Logical repository {repo} must be released either by tag or by "
-                                "submitted event."
-                            ),
-                        ),
-                    )
+            repo_name = RepositoryName.from_prefixed(repo)
+            if repo_name.is_logical and match not in (ReleaseMatch.tag, ReleaseMatch.event):
+                msg = (
+                    f"Logical repository {repo} must be released either by tag or by "
+                    "submitted event."
+                )
+                raise ResponseError(InvalidRequestError(f"{pointer_root}.{match}", detail=msg))
+            repo_id = prefixer.repo_name_to_node[repo_name.unprefixed_physical]
```

Comment:
If we catch a KeyError here, we will break with 500. We should handle this exception.

Reply:
We already handle this thanks to `resolve_repos` called with deference=True.
The only set_release_matches(deference=False) is done with a controlled repo name so I think we can avoid explicit handling